eng/scripts/update-pathversions.rs

```diff
@@ -10,14 +10,24 @@ toml_edit = "^0.22"
 ---
 
 
+
 use regex::Regex;
 use std::io::Write;
 use std::{env, error::Error, fs, path::PathBuf};
 use toml_edit::{value, DocumentMut, Item, Table};
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let script_root = PathBuf::from(env::var("CARGO_MANIFEST_DIR")?);
-    let repo_root = script_root.join("../../..").canonicalize()?;
+    let mode = env::args().nth(1).unwrap_or("update".to_string());
```

Comment:
Too much magic, IMO. Like most CLIs, I'd rather we fail and make them pass explicit verbs. Just `expect("requires 'add' or 'update' command")` or something instead of `unwrap_or(...)`.